### PR TITLE
test: Mark uncaught TypeErrors in NdefRecord

### PR DESCRIPTION
**Pull request by AiyionPrime**
Good afternoon;
I played around with type checking and found five paths, which lead to uncaught TypeErrors due to optional fields in `NdefRecord`: `type`, `id` and `payload`.

There are eleven read accesses falsely assuming these fields would not be `None`.
Six of them are being shadowed by the five I marked in the tests.

Are these indeed expected? Or should they be caught?

### CODE REVIEW

**Comment by kichik** [comment]
Sounds like we should initialize them to `b''`?

**Comment by AiyionPrime** [comment]
At least from type checking side this would resolve the issues. I'll add two commits accordingly for you to assess

**Comment by AiyionPrime** [comment]
The fixup commit is something I messed up in the tests; I'd squash it before merging.
The two remaining tests are raising these, which I'm not sure is expected behavior.

```python
InvalidNdefRecord('RTD_TEXT payload missing status byte')
InvalidNdefRecord('RTD_URI payload missing status byte')
```

**Review by kichik — APPROVED** [review]

**Comment by kichik** [comment]
I haven't worked with NDEF in years 😄 I am going to have to defer to my old self and say those are expected behavior. Kind of makes sense too. URI or text with no content sounds like a buggy NDEF.

Android source code also supports this:

https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/nfc/NdefRecord.java#759

```
private Uri parseWktUri() {
        if (mPayload.length < 2) {
            return null;
        }
```

**Comment by AiyionPrime** [comment]
I just hope I don't cause this repo more trouble than gain.
Will rebase and squash the fixup.

**Comment by kichik** [comment]
Thanks. Let me know when you want me to merge. I always squash, so no need for rebase or very detailed commits in the branch.

**Comment by AiyionPrime** [comment]
I'm fine with these commits; the next PR will contain my take at the typehints.

**Comment by kichik** [comment]
Why are the tests not automatically running on this PR? 🤔
Were you able to run all tests locally? Did they pass?

**Comment by AiyionPrime** [comment]
Not sure. When I do a PR against my own repo, respectively this branch it works:
https://github.com/AiyionPrime/pyndef/pull/1

**Comment by AiyionPrime** [comment]
Locally all tests succeeded.
(And the two expected to fail did.)

**Pull request merged**
